CONTRIBUTING.rst

```diff
@@ -1,37 +1,214 @@
 Contributing
 ============
 
-Running Tests
--------------
+Instructions for contributors
+-----------------------------
+
+In order to make a clone of the `GitHub <https://github.com/aio-libs/aiobotocore>`_ repo: open the link and press the
+"Fork" button on the upper-right menu of the web page.
+
+I hope everybody knows how to work with git and gitlab nowadays :)
+
+Workflow is pretty straightforward:
+
+  1. Clone the `GitHub <https://github.com/aio-libs/aiobotocore>`_ repo
+
+  2. Make a change
+
+  3. Make sure all tests passed
+
+  4. Make sure the coverage doesn't get worse.
+
+  5. Commit changes to own aiobotocore
+
+  6. Correct message for ``commit``. You need read (`Commit message format`_).
 
-.. _GitHub: https://github.com/aio-libs/aiobotocore
+  7. Correct name for ``branch``. You need read (`Branch name format`_).
 
-Thanks for your interest in contributing to ``aiobotocore``, there are multiple
-ways and places you can contribute.
+  8. Make pull request from github page for your branch against master branch
 
-Fist of all just clone repository::
+  9. Optionally make backport pull Request(s) for landing a bug fix
+     into released aiobotocore versions.
 
-    $ git clone git@github.com:aio-libs/aiobotocore.git
+  10. Add changes in file ``CHANGES.txt`` for this read (`Changelog update`_).
 
-Create virtualenv with at least python3.5 (older version are not supported).
-For example using *virtualenvwrapper* commands could look like::
+
+Preconditions for running aiobotocore test suite
+------------------------------------------------
+
+We expect you to use a python virtual environment to run our tests.
+
+There are several ways to make a virtual environment.
+
+If you like to use *virtualenv* please run:
+
+.. code-block:: shell
 
    $ cd aiobotocore
-   $ mkvirtualenv --python=`which python3.5` aiobotocore
+   $ virtualenv -p /usr/bin/python3 venv-aiobotocore
+   $ . venv-aiobotocore/bin/activate
+
+For standard python *venv-aiobotocore*:
+
+.. code-block:: shell
+
+   $ cd aiobotocore
+   $ python3 -m venv venv-aiobotocore
+   $ . venv-aiobotocore/bin/activate
+
+For *virtualenvwrapper*:
+
+.. code-block:: shell
+
+   $ cd aiobotocore
+   $ mkvirtualenv -p /usr/bin/python3 venv-aiobotocore
+   $ workon venv-aiobotocore
+
+There are other tools like *pyvenv* but you know the rule of thumb
+now: create a python3 virtual environment and activate it.
+
+After that please install libraries required for development:
+
+.. code-block:: shell
+
+   $ pip install -r requirements/dev.txt
+
+Congratulations, you are ready to run the test suite!
+
+Run aiobotocore test suite
+--------------------------
+
+After all the preconditions are met you can run tests typing the next
+command:
+
+.. code-block:: shell
+
+   $ make test
```

Comment:
again, would be nice to switch to tox

Reply:
for the `make test` you can hide just launch tests. it's just a level of abstraction